Taskfile.yml

```diff
@@ -8,10 +8,17 @@ vars:
     sh: echo $(go list ./... | tr '\n' ' ')
   PROJECT_NAME: "serial-discovery"
   DIST_DIR: "dist"
-  VERSION:
-    sh: echo "$(git describe --tags --dirty --broken)"
+  # build vars
+  COMMIT:
+    sh: echo "$(git log --no-show-signature -n 1 --format=%h)"
+  TAG:
+    sh: echo "$(git tag --points-at=HEAD 2> /dev/null | head -n1)"
+  VERSION: "{{if .NIGHTLY}}nightly-{{.TIMESTAMP_SHORT}}{{else if .TAG}}{{.TAG}}{{else}}{{.PACKAGE_NAME_PREFIX}}git-snapshot{{end}}"
   TIMESTAMP:
     sh: echo "$(date -u +"%Y-%m-%dT%H:%M:%SZ")"
+  TIMESTAMP_SHORT:
+    sh: echo "{{now | date "20060102"}}"
```

Comment:
```suggestion
  TIMESTAMP:
    sh: echo "$(date -u +"%Y-%m-%dT%H:%M:%SZ")"
  TIMESTAMP_SHORT:
    sh: echo "{{now | date "20060102"}}"
  VERSION: "{{if .NIGHTLY}}nightly-{{.TIMESTAMP_SHORT}}{{else if .TAG}}{{.TAG}}{{else}}{{.PACKAGE_NAME_PREFIX}}git-snapshot{{end}}"
```
The variable must be defined before being referenced. With the previous code:
```
per@HAL MINGW64 /e/electronics/git-nobackup/serial-discovery (scerza/go-release)
$ export NIGHTLY=true && task build && ./serial-discovery --version
←[32mtask: [build] go build -v -ldflags ' -X github.com/arduino/serial-discovery/version.Tag=nightly- -X github.com/arduino/serial-discovery/version.Timestamp=2021-08-09T01:36:44Z '

←[0mserial-discovery nightly- (build timestamp: 2021-08-09T01:36:44Z)
```
After this change:
```
per@HAL MINGW64 /e/electronics/git-nobackup/serial-discovery (scerza/go-release)
$ export NIGHTLY=true && task build && ./serial-discovery --version
←[32mtask: [build] go build -v -ldflags ' -X github.com/arduino/serial-discovery/version.Tag=nightly-20210808 -X github.com/arduino/serial-discovery/version.Timestamp=2021-08-09T01:37:46Z '

←[0mserial-discovery nightly-20210808 (build timestamp: 2021-08-09T01:37:46Z)
```

Reply:
Yep, I missed that. :+1: